src/java_bytecode/java_types.cpp

```diff
@@ -884,3 +884,54 @@ optionalt<size_t> java_generic_symbol_typet::generic_type_index(
   }
   return {};
 }
+
+std::string pretty_java_type(const typet &type)
+{
+  if(type==java_int_type())
+    return "int";
+  else if(type==java_long_type())
+    return "long";
+  else if(type==java_short_type())
+    return "short";
+  else if(type==java_byte_type())
+    return "byte";
+  else if(type==java_char_type())
+    return "char";
+  else if(type==java_float_type())
+    return "float";
+  else if(type==java_double_type())
+    return "double";
+  else if(type==java_boolean_type())
+    return "boolean";
+  else if(type==java_byte_type())
+    return "byte";
+  else if(type.id()==ID_symbol)
+    return id2string(strip_java_namespace_prefix(
+      to_symbol_type(type).get_identifier()));
+  else
+    return "?";
```

Comment:
what's with array types ? Shouldn't this case be `UNREACHABLE` ?

Reply:
done